cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -62,11 +105,139 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 // Update updates a machine and is invoked by the Machine Controller
 func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	glog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+	error := errorWrapper{cluster: cluster, machine: machine}
+
+	// TODO: Should we lookup by name if this fails?
+	libvirtStatus, err := ProviderStatusFromMachine(machine)
+	if err != nil {
+		return error.WithLog(err, "error getting provider status")
+	}
+
+	uuid, err := uuid.Parse(*libvirtStatus.InstanceID)
```

Comment:
This panics when `libvirtStatus.InstanceID` is `nil`. In case you call the `Update` from scratch without populating machine status.

Reply:
It should be fine to look based on domain name (to be define under `cloud/libvirt/actuators/machine/utils/domain.go`):
```go
func LookupDomainByName(machine *clusterv1.Machine) (*libvirt.Domain, error) {
	machineProviderConfig, err := MachineProviderConfigFromClusterAPIMachineSpec(&machine.Spec)
	client, err := buildClient(machineProviderConfig.URI)
	if err != nil {
		return nil, fmt.Errorf("Failed to build libvirt client: %s", err)
	}

	virConn := client.libvirt
	if virConn == nil {
		return nil, fmt.Errorf(LibVirtConIsNil)
	}

	dom, err := virConn.LookupDomainByName(machine.Name)
	if err != nil {
		return nil, fmt.Errorf("Failed to lookup domain by name: %v", err)
	}

	return dom, nil
}
```

Reply:
I ditched the whole UUID thing. It didn't make sense to look up the domain via a field stored in the status in order to update the status anyway. The domain is looked up by name now.